lamindb_setup/core/upath.py

```diff
@@ -421,6 +425,54 @@ def view_tree(
     logger.print(message)
 
 
+# TODO: replace with something faster at some point
+# also consolidate with what's in _settings_storage
+def get_bucket_region(bucket_name) -> str:
```

Comment:
Why is this needed? We already have a proper function for that https://github.com/laminlabs/lamindb-setup/blob/ef0f0fe29edb9dfabc7d51528ca0d7c90e32c5fb/lamindb_setup/core/_settings_storage.py#L29

Reply:
Your new variant looks problematic to me because it doesn't account for the case when no credentials are present in the system.

Reply:
Yes, I don't like this. It's AI-generated.